server/src/main/scala/services/crunch/Optimiser.scala

```diff
@@ -0,0 +1,78 @@
+package services.crunch
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.stream.Materializer
+import org.slf4j.{Logger, LoggerFactory}
+import services.{OptimizerConfig, TryRenjin}
+import upickle.default.{macroRW, ReadWriter => RW}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+
+trait OptimiserLike {
+  def uri: String
```

Comment:
should this be in the trait seeing as it's only required for the http service

Reply:
Very good point :)